okpt/io/config/validator.py

```diff
@@ -0,0 +1,103 @@
+"""Provides validation for various config files.
+
+Functions:
+    validate(): Validate the tool configuration.
+
+Exceptions:
+    ConfigurationError: An error in the tool configuration.
+
+"""
+
+import os
+from io import TextIOWrapper
+from typing import Any, Dict
+
+import cerberus
+
+from okpt.io import reader
+from okpt.io.config import parser
+
+_CURR_FILE_DIR = os.path.dirname(os.path.abspath(__file__))
+_SCHEMAS_DIR = os.path.join(_CURR_FILE_DIR, 'schemas')
+_TOOL, _OPENSEARCH = ['tool', 'opensearch']
+
+
+class ConfigurationError(Exception):
+    """Exception raised for errors in the tool configuration.
+
+    Attributes:
+        message -- explanation of the error
+    """
+    def __init__(self, message: str):
+        self.message = message
+
+
+class _Validator():
+    """Validator for various configuration schemas.
+
+    Attributes:
+        validator: Cerberus validator for a particular schema
+        errors: Cerberus validation errors (if any are found during validation)
+    """
+    def __init__(self, schema_name: str):
+        self.validator = self._get_validator_from_schema_name(schema_name)
+        self.errors = ''
+
+    def _get_validator_from_schema_name(self, schema_name: str):
+        """Get the corresponding Cerberus validator from a schema name."""
+        schema_file_path = os.path.join(_SCHEMAS_DIR, f'{schema_name}.yml')
+        schema_file_obj = reader.get_read_file_obj(schema_file_path)
+        schema_obj = parser.parse_yaml(schema_file_obj)
+        return cerberus.Validator(schema_obj)
+
+    def validate(self, config_obj: Dict[str, Any]):
+        """Validate the configuration obj against the class schema."""
+        if not self.validator.validate(config_obj):
+            self.errors = self.validator.errors
```

Comment:
So how would a user see these errors? Should validate throw an exception as opposed to just returning a boolean?

Reply:
The user of the `_Validator` class (which in this case is the `validate` method in the `validator.py` module) would call `_Validator.validate` and if it returned false, then it would raise an error. This is intended to follow a [similar pattern used in Cerberus](https://docs.python-cerberus.org/en/stable/usage.html).

>Unlike other validation tools, Cerberus will not halt and raise an exception on the first validation issue. The whole document will always be processed, and False will be returned if validation failed. You can then access the errors property to obtain a list of issues. See Errors & Error Handling for different output options.